### ISSUE TITLE
Missing newlines for the Wrapping rule is showing the whole file as broken

### ISSUE THREAD

**Issue by mkrussel77** [detekt/detekt-intellij-plugin#189]
The plugin is not reporting the correct location for the Wrapping warnings in the following example.  

```kotlin
package foo

/**
 * Test showing whole file listed for missing new line warning.
 */
public class MissingNewline {
    private fun calculateResult(): Result<Int> {
        return Result.success(5)
    }

    /**
     * Do some work.
     */
    public fun doSomething() {
        calculateResult().fold(onSuccess = {
            it
        },
            onFailure = {
                100
            })
    }
}
```

The warning panel shows.

> Warning:(1, 1) Detekt - Wrapping: Missing newline after "("
> Warning:(1, 1) Detekt - Wrapping: Missing newline before ")"

And the IDE renders the whole file as a warning.

<img width="1509" alt="warning_highlight" src="https://user-images.githubusercontent.com/14059294/175374943-2a6de323-56cd-47b1-9689-325c7d1962a2.png">

When I run detekt with gradle, I get the correct line numbers returned.

This is using `Android Studio 2021.1.1 Patch 2` on MacOS with Detekt Plugin version 1.20.1

**Comment by chao2zhang** [comment]
I was able to reproduce partially that detekt plugin is reporting the violations probably for the entire file, but with less aggressive UI:
![image](https://user-images.githubusercontent.com/1175468/175853495-9dee5a90-1dcd-4fcd-88b8-b902ba7f0260.png)

Environment:
- IntelliJ 2022.1
- Enable ktlint (formatting) rules on

**Comment by chao2zhang** [comment]
There was a past attempt to fix this issue for `Indentation` rule in https://github.com/detekt/detekt/pull/4030/. All ktlint rules with `RunOnRootNodeOnly` are impacted.

Solution 1 is that we need to expand the fix to `Wrapping` and `FinalNewline`.
Solution 2 is waiting for legacy intelliJ plugin versions to be deprecated, because as shown in the screenshot, the latest IntellIJ IDE has a less aggressive UI for file level finding.

**Comment by arturbosch** [comment]
> Solution 2 is waiting for legacy intelliJ plugin versions to be deprecated, because as shown in the screenshot, the latest IntellIJ IDE has a less aggressive UI for file-level finding. `FinalNewline` and `Filename` still remain as file level.

This was my PR at https://github.com/detekt/detekt-intellij-plugin/pull/174 trying to minimize whole file marking.
For many cases the PR already works. I'm wondering why for `Filename` and `MatchingDeclarationName` it does not.
I agree with Solution 1 for `Wrapping` .
